iota/api.py

```diff
@@ -22,948 +23,948 @@
+
+        :return:
+          Dict with the following structure::
+
+             {
+               'trytes': List[TransactionTrytes],
+                 List of TransactionTrytes that were broadcast.
+                 Same as the input ``trytes``.
+             }
+
+        References:
+          - https://github.com/iotaledger/wiki/blob/master/api-proposal.md#broadcastandstore
+        """  # noqa
+        return extended.BroadcastAndStoreCommand(self.adapter)(trytes=trytes)
+
+    def get_account_data(self, start=0, stop=None, inclusion_states=False):
+        # type: (int, Optional[int], bool) -> dict
+        """
+        More comprehensive version of :py:meth:`get_transfers` that returns
+        addresses and account balance in addition to bundles.
+
+        This function is useful in getting all the relevant information of
+        your account.
+
+        :param start:
+          Starting key index.
+
+        :param stop:
+          Stop before this index.
+          Note that this parameter behaves like the ``stop`` attribute in a
+          :py:class:`slice` object; the stop index is *not* included in the
+          result.
+
+          If ``None`` (default), then this method will check every address
+          until it finds one without any transfers.
+
+        :param inclusion_states:
+          Whether to also fetch the inclusion states of the transfers.
+
+          This requires an additional API call to the node, so it is
+          disabled by default.
+
+        :return:
+          Dict containing the following values::
+
+             {
+               'addresses': List[Address],
+                 List of generated addresses.
+                 Note that this list may include unused addresses.
+
+               'balance': int,
+                 Total account balance.  Might be 0.
+
+               'bundles': List[Bundle],
+                 List of bundles with transactions to/from this account.
+             }
+        """
+        return extended.GetAccountDataCommand(self.adapter)(
+            seed=self.seed,
+            start=start,
+            stop=stop,
+            inclusionStates=inclusion_states,
+        )
+
+    def get_bundles(self, transaction):
+        # type: (TransactionHash) -> dict
+        """
+        Returns the bundle(s) associated with the specified transaction
+        hash.
+
+        :param transaction:
+          Transaction hash.  Must be a tail transaction.
+
+        :return:
+          Dict with the following structure::
+
+             {
+               'bundles': List[Bundle],
+                 List of matching bundles.  Note that this value is always
+                 a list, even if only one bundle was found.
+             }
+
+        :raise:
+          - :py:class:`iota.adapter.BadApiResponse` if any of the
+            bundles fails validation.
+
+        References:
+          - https://github.com/iotaledger/wiki/blob/master/api-proposal.md#getbundle
+        """  # noqa
```

Comment:
❔ Do these need `# noqa`?

Reply:
Flake8 is really picky, I have to admint. line 535 has more than 80 chars... I think the picky-ness of flake can be tuned. I'll checkout if doc strings can be ignored, if `# noqa` is undesireable.

Reply:
Hm.  I guess that's fine.   think it's useful to check line length on docstrings in general; if flake8 doesn't want to make exceptions for URLs, then we can live with this.

Anyway, the only alternatives I can think of are pretty awful, comparatively (split the URL into multiple lines, use URL shortener).

Reply:
Yeah, I also think URLs should be kept the way they are.
I'm going to update this PR asap